test/test_xcpp_kernel.py

```diff
@@ -132,6 +133,43 @@ class XCppTests(jupyter_kernel_test.KernelTests):
             }
         ]
 
+    #Helper function to execute the notebooks
+    def execute_notebook(inp, out):
+
+        # Execute the notebook
+        executed_notebook = pm.execute_notebook(
+            inp,
+            out,
+            log_output=True,
+            kernel_name='xcpp20'
+        )
+
+        # Read the input and output notebooks
+        with open(inp) as f:
+            input_nb = nbformat.read(f, as_version=4)
+        with open(out) as f:
+            output_nb = nbformat.read(f, as_version=4)
+
+        check = True
+
+        # Iterate over the cells in the input and output notebooks
+        for input_cell, output_cell in zip(input_nb.cells, output_nb.cells):
+            if input_cell.cell_type == 'code' and output_cell.cell_type == 'code':
+                # Check the outputs of the cells
+                for input_output, output_output in zip(input_cell.outputs, output_cell.outputs):
+                    if input_output.get('text') != output_output.get('text'):
+                        check = False
+                        break
+        
+        return check
+
+    # Tests for Notebooks
+    class XCppNotebookTests(unittest.TestCase):
+
+        def test_simple_notebook(self):
```

Comment:
I thought we can generate that test function, too...

Reply:
Just like the other classes, if the user wants to check a particular notebook he will add a definition with appropriate name. If the test fails somewhere this will help us know where it did. Let me know if you want me to make this into a list and iterate it instead.

Reply:
Yes, it should be easy to add a new notebook and having a list is what we want I think.